index.js

```diff
@@ -160,6 +169,15 @@ module.exports = (options = {}) => {
           continue;
         }
         let parsedLicense = licenseObjects(parseSPDX(dependency.license));
+        if (whitelist) {
+          for (let license of flattenLicense(parsedLicense)) {
+            if (!whitelistSet.has(license)) {
```

Comment:
Is this being conservative in the sense that every mentioned license has to be whitelisted? I suppose we could have a more complex logical evaluator that handles the “MIT or GPL” case? (Doesn’t seem necessary to do that, just want to understand the intent…)

Reply:
Yeah, the intent is for this to be very conservative - I think there could be a more complex evaluator, but `MIT or GPL` is sufficiently rare, and the cases where there are 'viral or non-viral' clauses might have extra conditions that we'd want to manually review.